Added privacy lock option to security

gemdev111 opened the pull request:
Added privacy lock toggle to Security Scene
Added logic in LockWindowManager to support "privacy lock option" enabling/disabling
extended tests with a new logic

resolved #188

gemdev111 committed:
Added privacy lock option to security
Added logic to LockWindowManager to support "privacy lock option" enabling/disabling
Tests updates

gemdev111 committed:
naming updates

gemcoder21 commented on Gem/Core/Services/LockWindowManager.swift:
```diff
@@ -2,21 +2,36 @@
+    init(lockModel: LockSceneViewModel,
+         preferences: Preferences = .main) {
         self.lockModel = lockModel
+        self.preferences = preferences
```
let's inject `Preferences` into `LockSceneViewModel`

gemcoder21 commented on Packages/Store/Sources/Stores/PreferencesStore.swift:
```diff
@@ -69,7 +69,10 @@ public class PreferencesStore {
     public var isDeveloperEnabled: Bool
-    
+
+    @UserDefault(defaults: .standard, key: "is_privacy_lock_enabled", defaultValue: false)
```
any values related to security needs to be stored in the keychain

gemdev111 committed:
Removed PrivacyLock from Prefences and added to keychains
Updates app logic to support privacylock from keychains
Minor adjustments based on @gemcoder21 review

gemdev111 committed:
Merge branch 'main' into privacy-lock

gemdev111 committed:
updates MockKeystorePassword to fit protocol

gemdev111 committed:
Merge branch 'main'
